kingpin/actors/aws/cloudformation.py

```diff
@@ -230,6 +232,33 @@ def _discover_default_params(self, template_body):
         }
         return default_params
 
+    def _strip_hash(self, template_str: str, as_obj: bool = False):
+        """Strips the hash from the template.
+
+        Note: This will also strip the "Outputs" section if no other output exists. This might cause
+        issues when diffiing a template that contains an outputs section with no outputs.
+        """
+
+        template_obj = json.loads(template_str)
+
+        if len(KINGPIN_CFN_HASH_OUTPUT_KEY) > 0:
+            if "Outputs" not in template_obj.keys() or not isinstance(
+                template_obj["Outputs"], dict
+            ):
+                return template_str
+
+            template_outputs_keys = template_obj["Outputs"].keys()
+            if KINGPIN_CFN_HASH_OUTPUT_KEY in template_outputs_keys:
+                del template_obj["Outputs"][KINGPIN_CFN_HASH_OUTPUT_KEY]
+
+                if len(template_outputs_keys) == 0:
+                    del template_obj["Outputs"]
+
+        if as_obj:
+            return template_obj
```

Comment:
Don't do this inside the function. Make one function that returns the object, and then another function that calls the object-function and then stringifies it ... if necessary.